### PR TITLE
[MRG+1] Numpy Doc Format

### PR DESCRIPTION
**Pull request by sirlittle**
Changed some documentation to be in Numpy Doc format for set_xlim, set_ylim, etc. Also removed extra line from the documentation.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3796,7 +3796,7 @@ def dopatch(xs, ys, **kwargs):
     def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
                 vmin=None, vmax=None, alpha=None, linewidths=None,
                 verts=None, edgecolors=None,
-                **kwargs):
+                **kx`wargs):
```
Comment (thisch) [inline lib/matplotlib/axes/_axes.py]: accidentally commited?

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -2800,7 +2800,9 @@ def get_xlim(self):
         return tuple(self.viewLim.intervalx)
 
     def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
-        """
+        """ 
+        set_xlim(self, left, right, emit, auto, **kwargs)
```
Comment (tacaswell) [inline lib/matplotlib/axes/_base.py]: @NelleV haven't we been removing these because they are redundent?

**Comment by tacaswell** [comment]
Something funny went on with that merge from master.  Can you try re-basing this on top of current master See http://matplotlib.org/devdocs/devel/gitwash/development_workflow.html?highlight=rebase#rebasing-a-pull-request-pr

Please ping me if you need more help with rebasing.

File: lib/matplotlib/axes/_base.py (hunk 2 above)
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Yes, I removed as many as I could without losing information; please don't add them back.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -3796,7 +3796,7 @@ def dopatch(xs, ys, **kwargs):
     def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
                 vmin=None, vmax=None, alpha=None, linewidths=None,
                 verts=None, edgecolors=None,
-                **kwargs):
+                **kxwargs):
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: It's `kwargs`.

If possible, please squash that last commit (and any others modifying this line) into the second commit.

**Comment by NelleV** [comment]
This looks good, but next time please open a new pull request for each ticket you fix.
Can you also add the list of tickets this PR closes?

**Comment by sirlittle** [comment]
This pull request is supposed to solve #7205  and #7388.

File: doc/users/mathtext.rst [hunk 4]
```diff
@@ -216,7 +216,7 @@ When using the `STIX <http://www.stixfonts.org/>`_ fonts, you also have the choi
     ``\mathrm{\mathsf{sansserif}}``        :math-stix:`\mathrm{\mathsf{sansserif}}`
     ====================================== =========================================
 
-.. htmlonly::
+  .. htmlonly::
```
Comment (QuLogic) [inline doc/users/mathtext.rst]: So you've mostly reverted the first commit, so I'm a bit confused here.

File: lib/matplotlib/axes/_base.py [hunk 5]
```diff
@@ -2917,12 +2917,12 @@ def set_xscale(self, value, **kwargs):
         """
         Set the x-axis scale
 
-        Set the scaling of the x-axis: %(scale)s
-
-        ACCEPTS: [%(scale)s]
+        Parameters
+        ----------
+        value: float
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Missing space before the colon.

Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: I don't think this is true at all; it accepts the _strings_ defined [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/scale.py#L504).

File: lib/matplotlib/axes/_base.py [hunk 6]
```diff
@@ -2917,12 +2917,12 @@ def set_xscale(self, value, **kwargs):
...
+            Value to scale x-axis by
 
-        Different kwargs are accepted, depending on the scale:
-        %(scale_docs)s
+        Different kwargs are accepted, depending on the scale
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: I know @NelleV wants to remove this interpolation stuff, but right now you've reduced the information in the docstring. There needs to _at least_ be a cross-link to the place where the other keyword arguments are defined.

File: lib/matplotlib/axes/_base.py [hunk 7]
```diff
@@ -2995,15 +2995,21 @@ def get_xticklabels(self, minor=False, which=None):
     @docstring.dedent_interpd
     def set_xticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
-        Set the xtick labels with list of strings *labels*
+        Set the xtick labels with list of strings labels
 
-        Return a list of axis text instances.
+        Parameters
+        ----------
+        labels: list of str
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Missing space before colon.

File: lib/matplotlib/axes/_base.py [hunk 8]
```diff
@@ -2995,15 +2995,21 @@ def get_xticklabels(self, minor=False, which=None):
...
-        %(Text)s
+        Returns
+        -------
+        list of str
+            A list of axis text instances
 
-        ACCEPTS: sequence of strings
+        Other Parameters
+        -----------------
+        kwargs : :class:`~matplotlib.text.Text` properties.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Missing the `**`.

File: lib/matplotlib/axes/_base.py [hunk 9]
```diff
@@ -3196,14 +3201,18 @@ def get_yscale(self):
...
     @docstring.dedent_interpd
     def set_yscale(self, value, **kwargs):
-        """Set the y-axis scale
+        """
+        Set the y-axis scale
 
-        Set the scaling of the y-axis: %(scale)s
+        Parameters
+        ----------
+        value: float
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Missing space before colon, and just like `set_xscale`, I don't think the type is correct.

File: lib/matplotlib/axes/_base.py [hunk 10]
```diff
@@ -3196,14 +3201,18 @@ def get_yscale(self):
...
+            Value to scale y-axis by
 
-        ACCEPTS: [%(scale)s]
+        Other Parameters
+        -----------------
+        kwargs : :class:`~matplotlib.text.Text` properties.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Where did this come from?

File: lib/matplotlib/axes/_base.py [hunk 11]
```diff
@@ -3279,15 +3288,22 @@ def get_yticklabels(self, minor=False, which=None):
     @docstring.dedent_interpd
     def set_yticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
-        Set the y tick labels with list of strings *labels*
+        Set the xtick labels with list of strings labels
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: xtick -> ytick
strings -> string

File: lib/matplotlib/axes/_base.py [hunk 12]
```diff
@@ -2995,15 +2995,21 @@ def get_xticklabels(self, minor=False, which=None):
     @docstring.dedent_interpd
     def set_xticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
-        Set the xtick labels with list of strings *labels*
+        Set the xtick labels with list of strings labels
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: strings -> string

File: lib/matplotlib/axes/_base.py [hunk 13]
```diff
@@ -3279,15 +3288,22 @@ def get_yticklabels(self, minor=False, which=None):
     @docstring.dedent_interpd
     def set_yticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
-        Set the y tick labels with list of strings *labels*
+        Set the xtick labels with list of strings labels
+
+        Parameters
+        ----------
+        labels: list of str
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Missing space before colon.

File: lib/matplotlib/axes/_base.py [hunk 14]
```diff
@@ -2995,15 +2995,21 @@ def get_xticklabels(self, minor=False, which=None):
...
+            list of string labels
 
-        *kwargs* set the :class:`~matplotlib.text.Text` properties.
-        Valid properties are
-        %(Text)s
+        Returns
+        -------
+        list of str
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: The description does not say a list of str.

File: lib/matplotlib/axes/_base.py [hunk 15]
```diff
@@ -3279,15 +3288,22 @@ def get_yticklabels(self, minor=False, which=None):
...
+
+        Parameters
+        ----------
+        labels: list of str
+            list of string labels
 
-        Return a list of :class:`~matplotlib.text.Text` instances.
+        Returns
+        -------
+        list of str
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: The description says otherwise.

File: lib/matplotlib/axes/_base.py [hunk 16]
```diff
@@ -3279,15 +3288,22 @@ def get_yticklabels(self, minor=False, which=None):
...
+            A list of :class:`~matplotlib.text.Text` instances.
 
-        *kwargs* set :class:`~matplotlib.text.Text` properties for the labels.
-        Valid properties are
-        %(Text)s
+        Other Parameters
+        ----------------
+        *kwargs* set the :class:`~matplotlib.text.Text` properties.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: A different format than `set_xticklabels`.

File: lib/matplotlib/axes/_base.py [hunk 17]
```diff
@@ -3868,15 +3884,19 @@ def twinx(self):
         """
         Create a twin Axes sharing the xaxis
 
-        create a twin of Axes for generating a plot with a sharex
+        Create a twin of Axes for generating a plot with a sharex
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: sharex -> shared

File: lib/matplotlib/axes/_base.py [hunk 18]
```diff
@@ -3868,15 +3884,19 @@ def twinx(self):
...
 
-        .. note::
-            For those who are 'picking' artists while using twinx, pick
-            events are only called for the artists in the top-most axes.
+        Returns
+        -------
+        Axis
+            The newly created axis
+
+        For those who are 'picking' artists while using twinx, pick
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Needs a `Notes` heading.

File: lib/matplotlib/axes/_base.py [hunk 19]
```diff
@@ -3891,14 +3911,18 @@ def twiny(self):
...
 
-        .. note::
-            For those who are 'picking' artists while using twiny, pick
-            events are only called for the artists in the top-most axes.
+        Returns
+        -------
+        Axis
+            The newly created axis
+
+        For those who are 'picking' artists while using twiny, pick
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Needs a `Notes` heading.

**Comment by NelleV** [comment]
Hi @sirlittle 
Don't worry about the last elements to be done on this PR. I'll fix them.
On the other hand, please do not push anything onto your remote master branch until this is merged.
Thanks,

File: lib/matplotlib/axes/_base.py (hunk 8 above)
Comment (NelleV) [inline lib/matplotlib/axes/_base.py]: This is formatted properly.

Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: It's not about formatting as bold, but the double-asterisks to denote keyword arguments.

**Comment by NelleV** [comment]
I suspect the failing test here is unrelated to the patch but it is a test that is failing consistently.

File: lib/matplotlib/axes/_base.py [hunk 20]
```diff
@@ -2927,17 +2927,29 @@ def get_xscale(self):
...
 
-    @docstring.dedent_interpd
     def set_xscale(self, value, **kwargs):
         """
         Set the x-axis scale
 
-        Set the scaling of the x-axis: %(scale)s
+        Parameters
+        ----------
+        value : ["linear", "log", "symlog", "logit"]
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Should be braces.

File: lib/matplotlib/axes/_base.py [hunk 21]
```diff
@@ -3209,16 +3224,29 @@ def get_yscale(self):
...
 
-    @docstring.dedent_interpd
     def set_yscale(self, value, **kwargs):
-        """Set the y-axis scale
+        """
+        Set the y-axis scale
+
+        Parameters
+        ----------
+        value : ["linear", "log", "symlog", "logit"]
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: Braces here, too.

File: lib/matplotlib/axes/_base.py [hunk 22]
```diff
@@ -3291,18 +3319,22 @@ def get_yticklabels(self, minor=False, which=None):
                                  self.yaxis.get_ticklabels(minor=minor,
                                                            which=which))
 
-    @docstring.dedent_interpd
     def set_yticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
-        Set the y tick labels with list of strings *labels*
+        Set the xtick labels with list of strings labels
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: x -> y

Comment (NelleV) [inline lib/matplotlib/axes/_base.py]: Nice catch.

**Comment by NelleV** [comment]
Comments have been addressed (and I rebase against matplotlib/master and pushed force…).

**Comment by tacaswell** [comment]
Thanks!

**Pull request merged**